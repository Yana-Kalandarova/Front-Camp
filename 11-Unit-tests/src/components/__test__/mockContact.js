const mockContact = {
  name: 'Test Name',
  phoneNumber: '12345',
  _id: '1a2b3c',
};

export default mockContact;
