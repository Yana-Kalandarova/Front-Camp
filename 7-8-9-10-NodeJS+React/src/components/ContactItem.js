import React from 'react';
import PropTypes from 'prop-types';
import EditButton from './EditButton';
import DeleteButton from './DeleteButton';

const propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  onDeleteContact: PropTypes.func.isRequired,
};

const ContactItem = ({
  name, phoneNumber, _id, onDeleteContact,
}) => (
  <li className="contact_list-item">
    <dl>
      <dt>{name}</dt>
      <dd>{phoneNumber}</dd>
    </dl>
    <EditButton contactId={_id} />
    <DeleteButton onDeleteContact={onDeleteContact} contactId={_id} />
  </li>
);

ContactItem.propTypes = propTypes;

export default ContactItem;
