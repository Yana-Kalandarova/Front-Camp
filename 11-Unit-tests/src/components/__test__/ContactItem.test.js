import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import ContactItem from '../ContactItem';
import mockContact from './mockContact';

describe('ContactItem', () => {
  it('renders correctly', () => {
    expect(
      mount(
        <BrowserRouter>
          <ContactItem {...mockContact} onDeleteContact={jest.fn()}/>
        </BrowserRouter>
      )
    ).toMatchSnapshot()
  })
})
