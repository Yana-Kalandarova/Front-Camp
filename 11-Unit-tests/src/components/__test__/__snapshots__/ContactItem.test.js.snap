// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ContactItem
      _id="1a2b3c"
      name="Test Name"
      onDeleteContact={[MockFunction]}
      phoneNumber="12345"
    >
      <li
        className="contact_list-item"
      >
        <dl>
          <dt>
            Test Name
          </dt>
          <dd>
            12345
          </dd>
        </dl>
        <EditButton
          contactId="1a2b3c"
        >
          <Link
            className="contact_btn-edit"
            to="/edit-contact/1a2b3c"
          >
            <LinkAnchor
              className="contact_btn-edit"
              href="/edit-contact/1a2b3c"
              navigate={[Function]}
            >
              <a
                className="contact_btn-edit"
                href="/edit-contact/1a2b3c"
                onClick={[Function]}
              >
                <svg>
                  <use
                    xlinkHref="#undefined"
                  />
                </svg>
              </a>
            </LinkAnchor>
          </Link>
        </EditButton>
        <DeleteButton
          contactId="1a2b3c"
          onDeleteContact={[MockFunction]}
        >
          <button
            className="contact_btn-delete"
            onClick={[Function]}
            type="button"
          >
            <svg>
              <use
                xlinkHref="#undefined"
              />
            </svg>
          </button>
        </DeleteButton>
      </li>
    </ContactItem>
  </Router>
</BrowserRouter>
`;
